### DISCUSSION TITLE
Using Animate Parameter

### DISCUSSION THREAD

**Discussion by mattkg0** [cables-gl/cables_docs#827] [Answers & Questions]
Hi guys,

I've recently stumbled across cables.gl and firstly I would like to thank the developers. This is such a cool tool to play around with.

The ability to be able to animate any parameter with a key/value timeline I think could be very powerful, but I'm struggling to find much documentation on exactly how to use this. 

Is this feature complete for the time being or is it still a work in progress? Is it used widely across your patches or is it more common practice to drive animations using other ops instead?

What I've worked out so far for my test patch here:
https://cables.gl/edit/MmGU1l

Is that any parameter can be driven by an animation by clicking on the vertical dots of the parameter and clicking "Animate Parameter". From there you can add key frames along the timeline, you can then set the value for the key by selecting and typing in the value in the inspector. You can change the length (in frames) of the animation by clicking the "set timeline length" button.

Somethings I'm not sure are working are:
- The toggle loop button mostly works, but there is no visual feedback that animation is looping even though it is. It would awesome if the cursor looped back to the beginning of the animation if it loops to the end instead of continuing on.
- The options brought up with the timing meta panel need a bit of experimentation. I was about to say none of the easings seem to work, but I just realized you need to select multiple keys to use it.

Once again this is really great work. I'm really looking forward to continuing to use cables.gl and seeing it evolve. I just couldn't find anything in the existing documentation or forums covering the animate parameter stuff so thought I would post this to confirm and hopefully help someone else.

**Comment by 60-hz** [comment]
I agree that the timeline needs some work.
I started an issue about it many times ago https://github.com/cables-gl/cables_docs/issues/740
